azure/Kqlmagic/adal_token_cache.py

```diff
@@ -44,11 +56,33 @@ def _get_cache_key(entry):
 
 
 class AdalTokenCache(object):
-    def __init__(self, state=None):
-        ip = get_ipython()  # pylint: disable=undefined-variable
+    def __init__(self, user_id, encryption_key,salt, state=None):
+        ip = get_ipython()  # pylint: disable=E0602
         self.db = ip.db
         self._cache = {}
         self._lock = threading.RLock()
+
+        self.user_id = user_id
+        self.salt  = salt
+        self.encryption_key = encryption_key
```

Comment:
I suggest that the ecryption_key will be the concatination of the secret with the username. So that if we have two different usernames with same secret and same salt the result will be different encryption.